client/src/java/com/zimbra/client/ZContactHit.java

```diff
@@ -351,4 +356,39 @@ private String get(Map<String,String> attrs, String key, String defaultValue) {
         String value = attrs != null ? attrs.get(key) : null;
         return value != null ? value : defaultValue;
     }
+
+    @Override
+    public int getItemId() throws ServiceException {
+        return new ItemIdentifier(mId,  null).id;
+    }
+
+    @Override
+    public int getParentId() throws ServiceException {
+        return -1;
```

Comment:
Yeah I was actually wondering about this myself, but it seems like when you create a contact the parent is set to -1. There may be times when this is not the case, but for IMAP I don't think it really matters.